include/thread_pool/mpmc_bounded_queue.hpp

```diff
@@ -169,7 +208,33 @@ inline MPMCBoundedQueue<T>& MPMCBoundedQueue<T>::operator=(MPMCBoundedQueue&& rh
 
 template <typename T>
 template <typename U>
-inline bool MPMCBoundedQueue<T>::push(U&& data)
+inline bool MPMCBoundedQueue<T>::pushStrong(U&& data)
+{
+    return push(std::forward<U>(data), true);
+}
+
+template <typename T>
+inline bool MPMCBoundedQueue<T>::popStrong(T& data)
+{
+    return pop(data, true);
+}
+
+template <typename T>
+template <typename U>
+inline bool MPMCBoundedQueue<T>::pushWeak(U&& data)
+{
+    return push(std::forward<U>(data), false);
+}
+
+template <typename T>
+inline bool MPMCBoundedQueue<T>::popWeak(T& data)
+{
+    return pop(data, false);
+}
+
+template <typename T>
+template <typename U>
+inline bool MPMCBoundedQueue<T>::push(U&& data, bool is_strong)
 {
     Cell* cell;
     size_t pos = m_enqueue_pos.load(std::memory_order_relaxed);
```

Comment:
Strongly suggest having acquire memory ordering here. Because of reordering and predictive optimization some further code may execute before the load finishes, which is not what you desire here.

Reply:
Fixed in #3.